tests/backup/backup_test.go

```diff
@@ -1011,467 +1025,765 @@ var _ = Describe("{BackupCrashVolDriver}", func() {
+						destinationClusterName, orgID)
+					if err != nil {
+						logrus.Errorf("Failed to create restore [%s] in org [%s] on cluster [%s]. Error: [%v]",
+							restoreName, orgID, clusterName, err)
+						aliveRestore[namespace] = false
+					}
+				})
+			}
+			for _, namespace := range bkpNamespaces {
+				if !aliveRestore[namespace] {
+					continue
+				}
+				restoreName := fmt.Sprintf("%s-%s-%d", restoreNamePrefix, namespace, counter)
+				Step(fmt.Sprintf("Wait for restore %s:%s to complete",
+					namespace, restoreName), func() {
+					ctx, err := backup.GetPxCentralAdminCtx()
+					if err != nil {
+						logrus.Errorf("Failed to fetch px-central-admin ctx: [%v]", err)
+						aliveRestore[namespace] = false
+					} else {
+						err = Inst().Backup.WaitForRestoreCompletion(ctx, restoreName, orgID,
+							backupRestoreCompletionTimeoutMin*time.Minute,
+							retrySeconds*time.Second)
+						if err == nil {
+							logrus.Infof("Restore [%s] completed successfully", restoreName)
+							successfulRestores++
+						} else {
+							logrus.Errorf("Failed to wait for restore [%s] to complete. Error: [%v]",
+								restoreName, err)
+							aliveRestore[namespace] = false
+						}
+					}
+				})
+			}
+			for namespace, alive := range aliveBackup {
+				if alive {
+					backupName := fmt.Sprintf("%s-%s-%d", backupNamePrefix, namespace, counter)
+					successfulBackupNames = append(successfulBackupNames, backupName)
+				}
+			}
+			remainingContexts := make([]*scheduler.Context, 0)
+			for namespace, alive := range aliveRestore {
+				if alive {
+					restoreName := fmt.Sprintf("%s-%s-%d", restoreNamePrefix, namespace, counter)
+					successfulRestoreNames = append(successfulRestoreNames, restoreName)
+					for _, ctx := range namespaceContextMap[namespace] {
+						remainingContexts = append(remainingContexts, ctx)
+					}
+				}
+			}
+			// Change namespaces to restored apps only after backed up apps are cleaned up
+			// to avoid switching back namespaces to backup namespaces
+			Step("Validate Restored applications", func() {
+				destClusterConfigPath, err := getDestinationClusterConfigPath()
+				Expect(err).NotTo(HaveOccurred(),
+					fmt.Sprintf("Failed to get kubeconfig path for destination cluster. Error: [%v]", err))
 
-	log.Printf("Create creds for azure")
-	tenantID = os.Getenv("AZURE_TENANT_ID")
-	Expect(tenantID).NotTo(Equal(""),
-		"AZURE_TENANT_ID Environment variable should not be empty")
+				SetClusterContext(destClusterConfigPath)
 
-	clientID = os.Getenv("AZURE_CLIENT_ID")
-	Expect(clientID).NotTo(Equal(""),
-		"AZURE_CLIENT_ID Environment variable should not be empty")
+				// Populate contexts
+				for _, ctx := range remainingContexts {
+					ctx.SkipClusterScopedObject = true
+					ctx.SkipVolumeValidation = true
+				}
+				ValidateRestoredApplications(remainingContexts, volumeParams)
+			})
+			Step("teardown all restored apps", func() {
```

Comment:
Check if test exits before `Teardown` else check for errors before running `Teardown`.